catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1147,6 +1155,41 @@ public CreateResponse create(CreateRequest createRequest)
         return createResponse;
     }
 
+    private void applyAttributeOverridesToMetacardMap(String attributeOverrideHeaders,
+            Map<String, Metacard> metacardMap) {
+
+        Map<String, String> attributeOverrideMap = createAttributeOverrideMapFromHeaders(
+                attributeOverrideHeaders);
+
+        if (MapUtils.isEmpty(attributeOverrideMap) || MapUtils.isEmpty(metacardMap)) {
+            return;
+        }
+
+        attributeOverrideMap.forEach((attributeMapKey, attributeMapValue) -> metacardMap.forEach((metacardMapKey, metacardMapValue) -> {
+            if (metacardMapValue.getMetacardType()
+                    .getAttributeDescriptor(attributeMapKey) != null) {
+                metacardMapValue.setAttribute(new AttributeImpl(attributeMapKey,
```

Comment:
It doesn't appear we ever check if these are valid attributes. We need to check that they are injecting real attributes with the correct types.